src/arch/riscv64/exceptions.rs

```diff
@@ -0,0 +1,22 @@
+use core::convert::TryFrom;
+use num_enum::TryFromPrimitive;
+
+/// Enum of the different exceptions. See [Volume 2, Privileged Spec v.
+/// 20190608](https://github.com/riscv/riscv-isa-manual/releases/download/Ratified-IMFDQC-and-Priv-v1.11/riscv-privileged-20190608.pdf)
+///          table 4.2
+#[repr(usize)]
+#[derive(Debug, Eq, PartialEq, TryFromPrimitive)]
+enum Exception {
+    IllegalInstruction = 2,
+}
+
+/// Handle exceptions
+pub fn handle(id: usize, pc: usize) {
+    let exception = Exception::try_from(id).unwrap();
+    match exception {
+        Exception::IllegalInstruction => {
+            panic!("Illegal instruction at {:#x}!", pc)
+        }
+        _ => panic!("Exception {} not handled yet!. Occured at {:#x}", id, pc),
```

Comment:
I think that using `unreachable!` instead of `panic!` here removes the warning. If it does not, then keep it as is because it is important to have a "default" handler even when it produces warnings (for now) and that's good. Of course we'll remove it once all the different exceptions are added

Reply:
f7368eb Is ok for you?

Reply:
Ooooh, that's very nice. I didn't even know you could do that. Great idea 👍